rkt/status.go

```diff
@@ -16,22 +16,180 @@
 
 package main
 
-import "fmt"
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/appc/spec/schema/types"
+	"github.com/coreos/rocket/pkg/lock"
+)
 
 var (
 	cmdStatus = &Command{
-		Name:    "status",
-		Summary: "Check the status of a rkt job",
-		Usage:   "UUID",
+		Name:    statusCmd,
+		Summary: "Check the status of a rkt container",
+		Usage:   "[--wait] UUID",
 		Run:     runStatus,
 	}
+	flagWait bool
+)
+
+const (
+	statusDir = "stage1/rkt/status"
+	statusCmd = "status"
 )
 
 func init() {
 	commands = append(commands, cmdStatus)
+	cmdStatus.Flags.BoolVar(&flagWait, "wait", false, "toggle waiting for the container to exit if running")
 }
 
 func runStatus(args []string) (exit int) {
-	fmt.Println("Not implemented.")
+	if len(args) != 1 {
+		printCommandUsageByName(statusCmd)
+		return 1
+	}
+
+	containerUUID, err := types.NewUUID(args[0])
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid UUID: %v\n", err)
+		return 1
+	}
+
+	l, exited, err := getContainerLockAndState(containerUUID)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to access container: %v\n", err)
+		return 1
+	}
+	defer l.Close()
+
+	// There's a window between opening the container directory and lock acquisition where gc rename could occur,
+	// perform all subsequent opens relative to the opened container directory lock fd
+	cfd, err := l.Fd()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to get lock fd: %v\n", err)
+		return 1
+	}
+
+	if err = printStatusAt(cfd, exited); err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to print status: %v\n", err)
+		return 1
+	}
+
+	return 0
+}
+
+// getContainerLockAndState opens the container directory in the form of a lock.DirLock,
+// returning the lock and wether the container has already exited or not.
+func getContainerLockAndState(containerUUID *types.UUID) (l *lock.DirLock, isExited bool, err error) {
```

Comment:
yeah, some of these things need to be factored into a rkt/containers.go for reuse, just want to get the details of synchronization/implementation sorted.